applications/StructuralMechanicsApplication/custom_constitutive/constitutive_laws_integrators/high_cycle_fatigue_law_integrator.h

```diff
@@ -89,47 +89,33 @@ class HighCycleFatigueLawIntegrator
     ///@{
 
     /**
-     * @brief This method integrates checks if the uniaxial stress value of the previous step was a maximum or a minimum 
-     * @param CurrentStress The signed uniaxial stress in the current step.
-     * @param MaximumStress Identifies if the stress in the previous step was a maximum (signed uniaxial stress) in the calculation. Otherwise it is 0.
-     * @param MinimumStress Identifies if the stress in the previous step was a minimum (signed uniaxial stress) in the calculation. Otherwise it is 0.
-     * @param PreviousMaximumStress Latest identified maximum (signed uniaxial stress) in the calculation.
-     * @param PreviousMinimumStress Latest identified minimum (signed uniaxial stress) in the calculation.
-     * @param PreviousStresses Vector containing the stress value (signed uniaxial stresses) from the two previous steps.
-     * @param NumberOfCycles Number of cycles for the applied load.
-     * @param CycleCounter Boolean variable used on the update operation for the SetMaxStress.
+     * @brief This method checks and saves the previous stress state if it was a maximum or a minimum.
+     * @param CurrentStress Equivalent stress in the current step.
+     * @param rMaximumStress Maximum stress. 
+     * @param rMinimumStress Minimum stress. 
+     * @param PreviousStresses Equivalent stresses in the two previous steps. 
+     * @param rMaxIndicator Indicator of a maximum in the current cycle. 
+     * @param rMinIndicator Indicator of a minimum in the current cycle. 
      */
     static void CalculateMaximumAndMinimumStresses(
         const double CurrentStress,
-        double &rMaximumStress, 
-        double &rMinimumStress,  
-        double &rPreviousMaximumStress,
-        double &rPreviousMinimumStress,
-        const Vector &rPreviousStresses,
-        unsigned int& rNumberOfCycles,
-        bool& rCycleCounter
-        )
+        double& rMaximumStress, 
+        double& rMinimumStress,  
+        const Vector PreviousStresses,
+        bool& rMaxIndicator,
+        bool& rMinIndicator)
     {
-        const double stress_1 = rPreviousStresses[1];
-        const double stress_2 = rPreviousStresses[0];
+        const double stress_1 = PreviousStresses[1];
+        const double stress_2 = PreviousStresses[0];
         const double stress_increment_1 = stress_1 - stress_2;
         const double stress_increment_2 = CurrentStress - stress_1;
 
-        if (stress_increment_1 >= 0.001 && stress_increment_2 <= 0.0 && !rCycleCounter) {
+        if (stress_increment_1 >= 0.001 && stress_increment_2 <= 0.0) {
```

Comment:
I don't like the hardcoded coefficients 
```suggestion
        if (stress_increment_1 >= 1.0e-3 && stress_increment_2 <= 0.0) {
```